pkg/kubectl/run.go

```diff
@@ -281,8 +317,31 @@ func (BasicPod) Generate(genericParams map[string]interface{}) (runtime.Object,
 			pod.Spec.Containers[0].Args = args
 		}
 	}
+
+	if len(envs) > 0 {
+		pod.Spec.Containers[0].Env = envs
+	}
+
 	if err := updatePodPorts(params, &pod.Spec); err != nil {
 		return nil, err
 	}
 	return &pod, nil
 }
+
+func parseEnvs(envArray []string) ([]api.EnvVar, error) {
+	envs := []api.EnvVar{}
+	for _, env := range envArray {
+		if strings.Index(env, "=") != -1 {
```

Comment:
I don't think you need this and the error you return in the else clause as `len(parts) != 2` covers this.